logstash-core/lib/logstash/config/loader.rb

```diff
@@ -0,0 +1,90 @@
+require "logstash/config/defaults"
+
+module LogStash; module Config; module Loader
+  def self.format_config(config_path, config_string)
+    config_string = config_string.to_s
+    if config_path
+      # Append the config string.
+      # This allows users to provide both -f and -e flags. The combination
+      # is rare, but useful for debugging.
+      config_string = config_string + load_config(config_path)
+    else
+      # include a default stdin input if no inputs given
+      if config_string !~ /input *{/
+        config_string += LogStash::Config::Defaults.input
+      end
+      # include a default stdout output if no outputs given
+      if config_string !~ /output *{/
+        config_string += LogStash::Config::Defaults.output
+      end
+    end
+    config_string
+  end
+
+  def self.load_config(path)
```

Comment:
I know this was the previous name but can you change it to something more accurate like "fetch_config" and change fetch_config to...`from_url` or something?

Specially now that the loader is an isolated entity, It's strange that `LogStash::Config::Loader.load_config` is a stateless operation that retrieves a config from a path